src/deeponet.jl

```diff
@@ -0,0 +1,116 @@
+"""
+    DeepONet(; branch = (64, 32, 32, 16), trunk = (1, 8, 8, 16),
+        branch_activation = identity, trunk_activation = identity)
+
+constructs a DeepONet composed of Dense layers. Make sure the last node of `branch` and `trunk` are same.
+
+## Keyword arguments:
+
+- `branch`: Tuple of integers containing the number of nodes in each layer for branch net
+- `trunk`: Tuple of integers containing the number of nodes in each layer for trunk net
+- `branch_activation`: activation function for branch net
+- `trunk_activation`: activation function for trunk net
+
+## References:
+Lu Lu, Pengzhan Jin, George Em Karniadakis , "DeepONet: Learning nonlinear operators for identifying differential equations based on the universal approximation theorem of operators"
+doi: https://arxiv.org/abs/1910.03193
```

Comment:
JuliaFormatter currently has a shortcoming: being unable to format markdown in docstrings with the line limit. You need to manually format these ones.